### ISSUE TITLE
moving forward, changes are to be made to develop branch BEFORE being merged to public-v2

### ISSUE THREAD

**Issue by edwardhartnett** [NOAA-EMC/UFS_UTILS#294]
As I get a better understanding of the UFS_UTILS, I do see a big process problem.

Changes must be made to the develop branch before being ported back to the public-v2 release. The reason is that doing it in the other order:
* Builds up technical debt.
* Encourages development without unit testing.

It builds technical debt because it leaves the fix on develop for future work, and that will include testing, which will not have been done on the public-v2 branch. Yet once the public-v2 work is done, there will be little urgency to also make the fix on develop, which will lead to future bugs.

So, moving forward, the process is:
1 - Enter an issue for the bug/change.
2 - Fix on the develop branch, including full unit testing and documentation.
3 - Back-port changes to the public-v2 branch.

This matches the industry-standard process used everywhere else.
